Mathlib/Data/Matrix/Invertible.lean

```diff
@@ -171,6 +172,43 @@ theorem invOf_add_mul_mul [Invertible (A + U*C*V)] :
   letI := invertibleAddMulMul A U C V
   convert (rfl : ⅟(A + U*C*V) = _)
 
+variable {ι : Type*} [Unique ι]
+variable (u v : n → α)
+
+theorem add_col_mul_row_mul_invOf_eq_one
+    [Invertible (1 + row ι v * ⅟A * col ι u)] :
+    (⅟A - (⅟A * col ι u * ⅟(1 + row ι v * ⅟A * col ι u) * row ι v * ⅟A)) *
+      (A + col ι u * row ι v) = 1 := by
+  haveI : Invertible (1 : Matrix ι ι α) := invertibleOne
+  haveI : Invertible (⅟ 1 + row ι v * ⅟A * col ι u) := by simpa
+  simpa using add_mul_mul_invOf_mul_eq_one' A (col ι u) 1 (row ι v)
+
+/-- Like `Matrix.add_col_mul_row_mul_invOf_eq_one`, but with multiplication reversed. -/
+theorem add_col_mul_row_mul_invOf_eq_one'
+    [Invertible (1 + row ι v * ⅟A * col ι u)] :
+    (A + col ι u * row ι v) *
+      (⅟A - (⅟A * col ι u * ⅟(1 + row ι v * ⅟A * col ι u) * row ι v * ⅟A)) = 1 := by
+  haveI : Invertible (1 : Matrix ι ι α) := invertibleOne
+  haveI : Invertible (⅟ 1 + row ι v * ⅟A * col ι u) := by simpa
+  simpa using add_mul_mul_invOf_mul_eq_one A (col ι u) 1 (row ι v)
+
+variable (ι) in
+/-- If matrix `A` and the scalar `(1 + row ι v * ⅟A * col ι u)` are invertible,
+then so is (A + col ι u * row ι v) -/
+def invertibleAddColMulRow [Invertible (1 + row ι v * ⅟A * col ι u)] :
+    Invertible (A + col ι u * row ι v) where
+  invOf := (⅟A - (⅟A * col ι u * ⅟(1 + row ι v * ⅟A * col ι u) * row ι v * ⅟A))
+  invOf_mul_self := add_col_mul_row_mul_invOf_eq_one _ _ _
+  mul_invOf_self := add_col_mul_row_mul_invOf_eq_one' _ _ _
+
+/-- The **Sherman Morrison Rank-1 Update** (`⅟` version). -/
```

Comment:
Can you add the `⁻¹` version as well, as we have for the Woodbury identity (in another file)?